sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py

```diff
@@ -151,1047 +114,923 @@ class ServiceBusQueueTests(AzureMgmtTestCase):
-        queue_client = QueueClient.from_connection_string(
-            servicebus_namespace_connection_string,
-            name=servicebus_queue.name,
-            debug=False)
-
-        async with queue_client.get_sender() as sender:
-            for i in range(10):
-                message = Message("{}".format(i))
-                await sender.send(message)
+        async with ServiceBusClient.from_connection_string(
+            servicebus_namespace_connection_string, logging_enable=False) as sb_client:
 
-        renewer = AutoLockRenew()
-        messages = []
-        async with queue_client.get_receiver(idle_timeout=5, mode=ReceiveSettleMode.PeekLock, prefetch=10) as receiver:
-            async for message in receiver:
-                if not messages:
-                    messages.append(message)
-                    assert not message.expired
-                    renewer.register(message, timeout=60)
-                    print("Registered lock renew thread", message.locked_until, datetime.now())
-                    await asyncio.sleep(50)
-                    print("Finished first sleep", message.locked_until)
-                    assert not message.expired
-                    await asyncio.sleep(25)
-                    print("Finished second sleep", message.locked_until, datetime.now())
-                    assert message.expired
-                    try:
-                        await message.complete()
-                        raise AssertionError("Didn't raise MessageLockExpired")
-                    except MessageLockExpired as e:
-                        assert isinstance(e.inner_exception, AutoLockRenewTimeout)
-                else:
-                    if message.expired:
-                        print("Remaining messages", message.locked_until, datetime.now())
+            async with sb_client.get_queue_sender(servicebus_queue.name) as sender:
+                for i in range(10):
+                    message = Message("{}".format(i))
+                    await sender.send(message)
+
+            renewer = AutoLockRenew()
+            messages = []
+            async with sb_client.get_queue_receiver(servicebus_queue.name, idle_timeout=5, mode=ReceiveSettleMode.PeekLock, prefetch=10) as receiver:
+                async for message in receiver:
+                    if not messages:
+                        messages.append(message)
+                        assert not message.expired
+                        renewer.register(message, timeout=60)
+                        print("Registered lock renew thread", message.locked_until, datetime.now())
+                        await asyncio.sleep(50)
+                        print("Finished first sleep", message.locked_until)
+                        assert not message.expired
+                        await asyncio.sleep(25)
+                        await asyncio.sleep(max(0,(message.locked_until - datetime.now()).total_seconds()))
```

Comment:
locked_until and the related time based functionality should all be UTC, not local time

Reply:
already updated